azure/services/managedclusters/managedclusters.go

```diff
@@ -184,240 +70,57 @@ func (s *Service) Reconcile(ctx context.Context) error {
-			ip, _, err := net.ParseCIDR(managedClusterSpec.ServiceCIDR)
-			if err != nil {
-				return fmt.Errorf("failed to parse service cidr: %w", err)
-			}
-			// HACK: set the last octet of the IP to .10
-			// This ensures the dns IP is valid in the service cidr without forcing the user
-			// to specify it in both the Capi cluster and the Azure control plane.
-			// https://golang.org/src/net/ip.go#L48
-			ip[15] = byte(10)
-			dnsIP := ip.String()
-			managedCluster.NetworkProfile.DNSServiceIP = &dnsIP
-		} else {
-			managedCluster.NetworkProfile.DNSServiceIP = managedClusterSpec.DNSServiceIP
-		}
-	}
-
-	for i := range managedClusterSpec.AgentPools {
-		pool := managedClusterSpec.AgentPools[i]
-		profile := converters.AgentPoolToManagedClusterAgentPoolProfile(pool)
-		*managedCluster.AgentPoolProfiles = append(*managedCluster.AgentPoolProfiles, profile)
-	}
-
-	if managedClusterSpec.AADProfile != nil {
-		managedCluster.AadProfile = &containerservice.ManagedClusterAADProfile{
-			Managed:             &managedClusterSpec.AADProfile.Managed,
-			EnableAzureRBAC:     &managedClusterSpec.AADProfile.EnableAzureRBAC,
-			AdminGroupObjectIDs: &managedClusterSpec.AADProfile.AdminGroupObjectIDs,
-		}
-	}
-
-	handleAddonProfiles(managedCluster, managedClusterSpec)
-
-	if managedClusterSpec.SKU != nil {
-		tierName := containerservice.ManagedClusterSKUTier(managedClusterSpec.SKU.Tier)
-		managedCluster.Sku = &containerservice.ManagedClusterSKU{
-			Name: containerservice.ManagedClusterSKUNameBasic,
-			Tier: tierName,
-		}
-	}
-
-	if managedClusterSpec.LoadBalancerProfile != nil {
-		managedCluster.NetworkProfile.LoadBalancerProfile = &containerservice.ManagedClusterLoadBalancerProfile{
-			AllocatedOutboundPorts: managedClusterSpec.LoadBalancerProfile.AllocatedOutboundPorts,
-			IdleTimeoutInMinutes:   managedClusterSpec.LoadBalancerProfile.IdleTimeoutInMinutes,
-		}
-		if managedClusterSpec.LoadBalancerProfile.ManagedOutboundIPs != nil {
-			managedCluster.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs = &containerservice.ManagedClusterLoadBalancerProfileManagedOutboundIPs{Count: managedClusterSpec.LoadBalancerProfile.ManagedOutboundIPs}
-		}
-		if len(managedClusterSpec.LoadBalancerProfile.OutboundIPPrefixes) > 0 {
-			managedCluster.NetworkProfile.LoadBalancerProfile.OutboundIPPrefixes = &containerservice.ManagedClusterLoadBalancerProfileOutboundIPPrefixes{
-				PublicIPPrefixes: convertToResourceReferences(managedClusterSpec.LoadBalancerProfile.OutboundIPPrefixes),
-			}
-		}
-		if len(managedClusterSpec.LoadBalancerProfile.OutboundIPs) > 0 {
-			managedCluster.NetworkProfile.LoadBalancerProfile.OutboundIPs = &containerservice.ManagedClusterLoadBalancerProfileOutboundIPs{
-				PublicIPs: convertToResourceReferences(managedClusterSpec.LoadBalancerProfile.OutboundIPs),
+		// Update control plane endpoint.
+		if managedCluster.ManagedClusterProperties != nil && managedCluster.ManagedClusterProperties.Fqdn != nil {
```

Comment:
Would it be unexpected for the properties to be nil? If so maybe we could add a log.

Reply:
not sure, I kept the same logic that was there before https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2168/files#diff-ea46b89333855bb06ff20351c648ad0a792c9fc36689b982e871b9f6fee2ef7aL366

Reply:
removed this and the pointer dereference